crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -638,8 +644,34 @@ fn pending_validator_exchange_rate(
         })
         .collect::<Vec<(IotaAddress, ObjectID, ObjectID, u64, bool)>>();
 
+    // Try to find for validator candidate and get its exchange rates
+    for (_, df) in state.get_dynamic_fields(
+        system_state_summary.validator_candidates_id,
+        None,
+        system_state_summary.validator_candidates_size as usize,
+    )? {
+        let raw_address =
+            String::deserialize(df.name.value).map_err(|err| IotaError::from(err.to_string()))?;
```

Comment:
This resolves into an `iota_json_rpc::error::Error`, through the `From<IotaError>` implementation. This is rather opaque and misleading, let alone since `For<&str> for IotaError` resolves into a `GenericAuthority` variant. 

I reckon the most proper variant to use here is `iota_json_rpc::error::Error::StateRead`, since if the value of this dynamic field is not a string then the state must be corrupted. `StateRead` can be converted from `iota_json_rpc::authority_state::StateReadError`, which in turn can be converted from `anyhow::Error`.

Reply:
After more careful analysis I found that the deserializing as `String` was unecessary, this is the new [solution](https://github.com/iotaledger/iota/pull/4313/commits/9cf72f246047bee46937cce23431ce6d5bf20a52#diff-0a10b5dd25f3cf15826ae9c8e8ab585ac0e76d35d7c0f84a9334f7ed722b5406R680-R685)